VoteSystem.Console/Program.cs

```diff
@@ -114,53 +114,63 @@ static void Main(string[] args)
+                                if (pollPolicyFailedChecks.Count > 0)
                                 {
-                                    foreach (var a in pollService.CheckAllPolicy(poll_temp_name, user_temp_id))
+                                    foreach (var a in pollPolicyFailedChecks)
                                     {
                                         Console.WriteLine(a.Value);
                                         Console.ReadLine();
                                     }
                                     break;
                                 }
-                                foreach (var a in pollRepository.GetChoices(poll_temp_name))
+
+                                Poll poll1 = pollRepository.Get(poll_temp_name);
+                                foreach (var a in poll1.Choices)
                                 {
                                     Console.WriteLine($"{a.Name} \n {a.Description} \n");
                                 }
                                 Console.WriteLine("Write what you choose:");
-                                List<string> allChoices = new List<string>();
+                                List<int> allChoices = new List<int>();
                                 if (poll1.MutlipleSelection == true)
                                 {
                                     string option = Console.ReadLine();
-                                    allChoices.Add(option);
+                                    allChoices.Add(poll1.GetChoiceByName(option).Id);
                                     Console.WriteLine("Do you want to choose smth more? (Y/N)");
                                     string multipleResponse = Console.ReadLine();
                                     while (multipleResponse == "Y")
                                     {
                                         option = Console.ReadLine();
-                                        allChoices.Add(option);
+                                        allChoices.Add(poll1.GetChoiceByName(option).Id);
                                         Console.WriteLine("Do you want to choose smth more? (Y/N)");
                                         multipleResponse = Console.ReadLine();
                                     }
                                 }
                                 else
                                 {
                                     string option = Console.ReadLine();
-                                    allChoices.Add(option);
+                                    allChoices.Add(poll1.GetChoiceByName(option).Id);
                                 }
-                                foreach (var a in allChoices)
-                                {
-                                    voteService.Vote(user_temp_id, pollRepository.GetChoices(poll_temp_name).FirstOrDefault(c => c.Name == a).Id);                                }
+
+                                voteService.Vote(allChoices);
```

Comment:
you've captured all required information from the "kinda UI layer" - now pass them to service layer "all-at-once" instead of doing it in the "loop", doing it "with inline filters" e.t.c.